goctl api swagger error

I encountered an error while using goctl to generate the document, but I cannot see any error message. The API file contains a lot of content and cannot locate the error

The content of the file involves business and cannot be provided here

```bash
F:\merla\merla-protocol>goctl api swagger --api ./api/server.api --dir .
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xac59a5]

goroutine 1 [running]:
github.com/zeromicro/go-zero/tools/goctl/api/spec.(*Tags).Get(...)
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/spec/tags.go:52
github.com/zeromicro/go-zero/tools/goctl/api/swagger.typeContainsTag({0x2?, 0x0?, {0x0?, 0xc0005873ec?}}, {{0xc0003e3db0, 0xa}, {0xc00064ed20, 0x1, 0x1}, {0x30b5820, ...}}, ...)
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/swagger.go:179 +0xe5
github.com/zeromicro/go-zero/tools/goctl/api/swagger.consumesFromTypeOrDef({0x24?, 0x63?, {0x0?, 0x5?}}, {0xc0002d6330?, 0x0?}, {0x1d99ef0, 0xc00060ce40})
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/contenttype.go:21 +0x146
github.com/zeromicro/go-zero/tools/goctl/api/swagger.spec2Path({_, _, {_, _}}, {{_}, {_, _, _}}, {{0x0}, {0xc0002d6330, ...}, ...})
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/path.go:83 +0x225
github.com/zeromicro/go-zero/tools/goctl/api/swagger.spec2Paths({0x10?, 0x8b?, {0x0?, 0x8?}}, {{0xc000708b08, 0x3}, {0xc000284d88, 0x14, 0x23}})
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/path.go:24 +0x39d
github.com/zeromicro/go-zero/tools/goctl/api/swagger.spec2Swagger(0xc00018c690)
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/swagger.go:32 +0x3e5
github.com/zeromicro/go-zero/tools/goctl/api/swagger.Command(0xc00015db00?, {0x1b0e68e?, 0x4?, 0x1b0e592?})
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/api/swagger/command.go:49 +0xb9
github.com/spf13/cobra.(*Command).execute(0xc000369808, {0xc0002cb4c0, 0x4, 0x4})
        F:/store/GOPATH/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xa94
github.com/spf13/cobra.(*Command).ExecuteC(0xc000393808)
        F:/store/GOPATH/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
        F:/store/GOPATH/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/zeromicro/go-zero/tools/goctl/cmd.Execute()
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/cmd/root.go:47 +0x6c
main.main()
        F:/store/GOPATH/pkg/mod/github.com/zeromicro/go-zero/tools/goctl@v1.8.4/goctl.go:12 +0x48
```

--- a/tools/goctl/api/spec/tags.go
+++ b/tools/goctl/api/spec/tags.go
@@ -49,6 +49,9 @@ func Parse(tag string) (*Tags, error) {
 
 // Get gets tag value by specified key
 func (t *Tags) Get(key string) (*Tag, error) {
+	if t == nil {
+		return nil, errTagNotExist
+	}
 	for _, tag := range t.tags {
 		if tag.Key == key {
 			return tag, nil
@@ -61,6 +64,9 @@ func (t *Tags) Get(key string) (*Tag, error) {
 // Keys returns all keys in Tags
 func (t *Tags) Keys() []string {
 	var keys []string
+	if t == nil {
+		return keys
+	}
 	for _, tag := range t.tags {
 		keys = append(keys, tag.Key)
 	}
@@ -69,5 +75,8 @@ func (t *Tags) Keys() []string {
 
 // Tags returns all tags in Tags
 func (t *Tags) Tags() []*Tag {
+	if t == nil {
+		return []*Tag{}
+	}
 	return t.tags
 }

--- a/tools/goctl/api/spec/tags.go
+++ b/tools/goctl/api/spec/tags.go
@@ -63,10 +63,10 @@ func (t *Tags) Get(key string) (*Tag, error) {
 
 // Keys returns all keys in Tags
 func (t *Tags) Keys() []string {
-	var keys []string
 	if t == nil {
-		return keys
+		return []string{}
 	}
+	var keys []string
 	for _, tag := range t.tags {
 		keys = append(keys, tag.Key)
 	}

--- a/tools/goctl/api/spec/tags_test.go
+++ b/tools/goctl/api/spec/tags_test.go
@@ -1,0 +1,68 @@
+package spec
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTags_Get(t *testing.T) {
+	tags := &Tags{
+		tags: []*Tag{
+			{Key: "json", Name: "foo", Options: []string{"omitempty"}},
+			{Key: "xml", Name: "bar", Options: nil},
+		},
+	}
+
+	tag, err := tags.Get("json")
+	assert.NoError(t, err)
+	assert.NotNil(t, tag)
+	assert.Equal(t, "json", tag.Key)
+	assert.Equal(t, "foo", tag.Name)
+
+	_, err = tags.Get("yaml")
+	assert.Error(t, err)
+
+	var nilTags *Tags
+	_, err = nilTags.Get("json")
+	assert.Error(t, err)
+}
+
+func TestTags_Keys(t *testing.T) {
+	tags := &Tags{
+		tags: []*Tag{
+			{Key: "json", Name: "foo", Options: []string{"omitempty"}},
+			{Key: "xml", Name: "bar", Options: nil},
+		},
+	}
+
+	keys := tags.Keys()
+	expected := []string{"json", "xml"}
+	assert.Equal(t, expected, keys)
+
+	var nilTags *Tags
+	nilKeys := nilTags.Keys()
+	assert.Empty(t, nilKeys)
+}
+
+func TestTags_Tags(t *testing.T) {
+	tags := &Tags{
+		tags: []*Tag{
+			{Key: "json", Name: "foo", Options: []string{"omitempty"}},
+			{Key: "xml", Name: "bar", Options: nil},
+		},
+	}
+
+	result := tags.Tags()
+	assert.Len(t, result, 2)
+	assert.Equal(t, "json", result[0].Key)
+	assert.Equal(t, "foo", result[0].Name)
+	assert.Equal(t, []string{"omitempty"}, result[0].Options)
+	assert.Equal(t, "xml", result[1].Key)
+	assert.Equal(t, "bar", result[1].Name)
+	assert.Nil(t, result[1].Options)
+
+	var nilTags *Tags
+	nilResult := nilTags.Tags()
+	assert.Empty(t, nilResult)
+}

--- a/tools/goctl/api/spec/tags_test.go
+++ b/tools/goctl/api/spec/tags_test.go
@@ -65,4 +65,4 @@ func TestTags_Tags(t *testing.T) {
 	var nilTags *Tags
 	nilResult := nilTags.Tags()
 	assert.Empty(t, nilResult)
-}git
+}

--- a/tools/goctl/api/spec/tags_test.go
+++ b/tools/goctl/api/spec/tags_test.go
@@ -65,4 +65,4 @@ func TestTags_Tags(t *testing.T) {
 	var nilTags *Tags
 	nilResult := nilTags.Tags()
 	assert.Empty(t, nilResult)
-}
+}git

